src/components/custom/signup.js

```diff
@@ -50,55 +50,45 @@ export const Appearances = {
     public: 'public',
     newSignup: 'newSignup',
 }
-class Signup extends Component {
-    state = {
-        email: '',
-        is_submitting: false,
-        email_error_msg: '',
-        submit_status: '',
-        submit_error_msg: '',
-    }
-
-    validateEmail = (email) => {
+const Signup = (props) => {
+    const [email, setEmail] = useState('')
+    const [is_submitting, setSubmitting] = useState(false)
+    const [email_error_msg, setEmailErrorMsg] = useState('')
+    const [submit_status, setSubmitStatus] = useState('')
+    const [submit_error_msg, setSubmitErrorMsg] = useState('')
+
+    const validateEmail = (emailAddress) => {
```

Comment:
```suggestion
    const validateEmail = (email_address) => {
```

We use snake_case for variables/arguements